### ISSUE TITLE
Breakpoints don't open source files in R 4.0 if code contains a backslash

### ISSUE THREAD

**Issue by jmcphers** [rstudio/rstudio#6854]
### System details

    RStudio Edition : Both
    RStudio Version : 1.3.947-1
    R Version: 4.0.0

### Steps to reproduce the problem

Create the following R script and saved it as e.g., `escaped.R`:

```r
f <- function() {
  x <- "Do you like newlines?\n"
  return(x)
}
```

Then:

1. Set a breakpoint on the first line of the function (the assignment of `x`)
2. Source the script
3. Run `f()` in the console.

### Describe the problem in detail

The breakpoint is hit and the code browser appears. It says "Debug location is approximate". Notice that the code is incorrect -- it has `\\n` in place of the `\n`.  

![image](https://user-images.githubusercontent.com/470418/81617899-20ef7780-939b-11ea-8465-eb20db85e3d1.png)

### Describe the behavior you expected

The original source file should be opened instead of the code browser.

![image](https://user-images.githubusercontent.com/470418/81617960-43819080-939b-11ea-8f36-932b64402497.png)

This is what happens in R < 4.0, and also what happens if you remove the escaped newline (`\n`) from the string. 

The problem appears to be a change in the way R 4.0 constructs source references; RStudio is now somehow getting an escaped copy. Because this copy differs from the contents of the file (`\n` is not `\\n`), RStudio thinks that the function run under the debugger differs from the one in the file, and opens a code browser for it.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jmcphers** [comment]
Derived from a StackOverflow post: https://stackoverflow.com/questions/61593190/debug-location-is-approximate-because-the-source-is-not-available-in-r-4-0-0/61740540#61740540 and accompanying community forum post: https://community.rstudio.com/t/debug-location-is-approximate-because-the-source-is-not-available-in-r-4-0-0-rstudio/65535

**Comment by jmcphers** [comment]
This appears to be due to a change in R 4.0 that affects how `deparse` works.

### R 3.6.2

```r
> g <- function () { "newline \n" }
> deparse(g, control = "useSource")
[1] "function () { \"newline \\n\" }"
```

### R 4.0.0

```r
> g <- function () { "newline \n" }
> deparse(g, control = "useSource")
[1] "function () { \"newline \\\\n\" }"
```

This only happens with the `useSource` control option to `deparse`.

**Comment by jmcphers** [comment]
This feels like a bug in R, because the deparsed function is over-escaped (i.e. if you parse it again you won't get the original function back but a mangled copy with two backslashes).

**Comment by jmcphers** [comment]
Submitted bug to R here: https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17800

**Comment by dfalty** [comment]
Verified fixed 1.3.1031
